### PR TITLE
Playground/didchat2

### PR DESCRIPTION
**Pull request by Arxcis**
```
$ didchat help

    Usage:
        didchat <path> init
        didchat <path> doc    
        didchat <path> did    

        didchat <path> connect  <other> <other did>

        didchat <path> send     <other> <message>       -->  <encrypted message>
        didchat <path> receive  <encrypted message>     -->  <other>: <message>

    Example - talk to self:
        didchat . init
        didchat . connect Self $(didchat . did)
        didchat . receive $(didchat . send Self "Hello myself!")

    Example - talk to peer:
        didchat jonas init
        didchat snorre init
        
        didchat snorre connect Jonas $(didchat jonas did)
        didchat jonas connect Snorre $(didchat snorre did)

        didchat jonas receive $(didchat snorre send Jonas "Hello Jonas. How are you?")
        didchat snorre receive $(didchat jonas send Snorre "Hi Snorre:) I have seen better days.")

```

### CODE REVIEW

File: playground/didchat2/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        .from(&from_did)
+        .to(&to_vec[..])
+        .timed(Some(3600))
+        .body(message.as_bytes())
+        .as_jwe(&didcomm_rs::crypto::CryptoAlgorithm::XC20P);
+
+    // 4. Seal/encrypt message using shared secret
+    let sealed_message = message
+        .seal(&shared_secret)
+        .unwrap(); // @unwrap!!
```
Comment (vongohren) [inline playground/didchat2/src/lib.rs:120]: Unwrap after seal? How does that work?

Comment (Arxcis) [reply playground/didchat2/src/lib.rs:120]: Haha. They are unrelated. This is the signature of the `seal()`-function:
```rs
 pub fn seal(self, ek: &[u8]) -> Result<String, Error>
```
The result of the `seal()`-function is wrapped inside a `Result`-container, so I have to `unwrap()` to get the actualt result. As @marni has mentioned, using `unwrap()` everywhere is an anti-pattern, which reflects my level of Rust-competence. I think the amount of `.unwrap()` in the code, is inversely proportional to my Rust-skills :laughing:

Comment (Arxcis) [reply playground/didchat2/src/lib.rs:120]: `.unwrap()` is an error-handling pattern. There are many ways to handle errors in Rust. `.unwrap()` is the lowest cost option for beginners, but also the most dangerous and useless one, as it does not provide much extra information to the user of the software, if something goes wrong.

One good thing about `.unwrap()`, is that it communicates well to other developers, that you spent 0 calories considering what should happen, if an error occurs.

File: playground/didchat2/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+    if !std::fs::metadata(seed_path(path)).is_ok() {
+        // 2. Generate seed
+        let mut csprng = rand_core::OsRng{};
+        let seed = ed25519_dalek::SecretKey::generate(&mut csprng);
+        let seed_bytes = seed.as_bytes();
+
+        // 3. Write seed to file
+        let mut file = std::fs::File::create(seed_path(path)).unwrap();
+        file.write(seed_bytes).unwrap();
```
Comment (gtl-hig) [inline playground/didchat2/src/lib.rs:40]: replace `unwrap` with a question mark `?`

File: playground/didchat2/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        Ok(format!("{} already exists", path))
+    }
+}
+
+fn doc(path: &str) -> Result<String, std::io::Error> {
+    use did_key::DIDCore;
+    use did_key::KeyMaterial;
+    
+    // 1. Read seed from file
+    let seed = std::fs::read(seed_path(path)).unwrap();
```
Comment (gtl-hig) [inline playground/didchat2/src/lib.rs:53]: replace `unwrap` with a question mark `?`

File: playground/didchat2/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+    // 2. Transform seed to a did-document
+    let public_private_keypair = did_key::Ed25519KeyPair::from_seed(&seed);
+    let public_only_keypair = did_key::from_existing_key::<did_key::Ed25519KeyPair>(&public_private_keypair.public_key_bytes(), None);
+    let doc = public_only_keypair.get_did_document(did_key::CONFIG_LD_PUBLIC);
+
+    // 3. Serialize did-document to json
+    let did_document = serde_json::to_string_pretty(&doc).unwrap();
```
Comment (gtl-hig) [inline playground/didchat2/src/lib.rs:61]: replace `unwrap` with a question mark `?`

File: playground/didchat2/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    // 3. Print did
+    Ok(format!("{}", did))
+}
+
+fn connect(path: &str, name: &str, did: &str) -> Result<String, std::io::Error> {
+    use std::io::Write;
+
+    // 2. Create 'name' -> 'did' mapping
+    let mut file = std::fs::File::create(name_path(path, name)).unwrap();
+    file.write(did.as_bytes()).unwrap();
```
Comment (gtl-hig) [inline playground/didchat2/src/lib.rs:85]: replace `unwrap` with a question mark `?`

Do it for all the IO operations that can throw an error.

**Review by gtl-hig — COMMENTED** [review]
Good job - I think I've lot my previous comments, but, the bottom line is to use a slightly better error handling and propagate errors to the caller - otherwise it looks fine.

**Pull request merged**